### PR TITLE
improvement/#890/clean_up_toast_notification_styles_and_allow_user_to_close_on-demand

### PR DESCRIPTION
**Pull request by evangelos-com**
Hello, this pull request is for #890.

| BEFORE  | AFTER |
| ------------- | ------------- |
| image  | image  |
| image | image  |

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -25,7 +25,7 @@
   </head>
 
   <body class="h-full">
-    <div id="notification-tray" class="fixed z-50 space-y-1 top-6 right-6"></div>
+    <div id="notification-tray" class="fixed z-50 space-y-1 top-6 right-10"></div>
```
Comment (evangelos-com) [inline app/views/layouts/application.html.erb:28]: Based on Figma, we should really move this further to the left, so it's visually inside the app container.  On my box, this was the highest class available.  Does anyone know why larger numbers are not available to us, or is it just my box?

File: app/views/shared/_notification.html.erb [hunk 2]
```diff
@@ -34,6 +34,9 @@
             <path d="M18 10C18 14.4183 14.4183 18 10 18C5.58172 18 2 14.4183 2 10C2 5.58172 5.58172 2 10 2C14.4183 2 18 5.58172 18 10ZM3.6 10C3.6 13.5346 6.46538 16.4 10 16.4C13.5346 16.4 16.4 13.5346 16.4 10C16.4 6.46538 13.5346 3.6 10 3.6C6.46538 3.6 3.6 6.46538 3.6 10Z" fill="#E5E5E5" />
             <circle class="origin-center -rotate-90 animate-[stroke-fill_5s_300ms_forwards]" stroke="#141414" stroke-opacity="0.4" r="7.2" cx="10" cy="10" stroke-dasharray="43.9822971503" stroke-dashoffset="43.9822971503" />
           </svg>
+          <div class="absolute -top-2 -right-2">
+            <%= lucide_icon "x", class: "w-5 h-5 p-0.5 hidden group-hover:inline-block border border-alpha-black-50 border-solid rounded-[10px] bg-white text-gray-400 cursor-pointer", data: { action: "click->element-removal#remove" } %>
```
Comment (evangelos-com) [inline app/views/shared/_notification.html.erb:38]: To get the `x` to look correctly I had to create a border with round corners.  I tried the `circle-x` icon, but I couldn't change the circle colour to match the [supplied video](https://github.com/maybe-finance/maybe/discussions/874#discussioncomment-9817770).

To get the close button to appear when you move the mouse over the notification, like the video, I hid it first then I showed it with `group-hover:inline-block`.

File: app/views/shared/_notification.html.erb [hunk 3]
```diff
@@ -5,13 +5,13 @@
...
-      class: "max-w-80 bg-white shadow-xs border border-alpha-black-50 border-solid py-3 px-4 rounded-[10px] text-sm font-medium flex gap-4 #{animation}",
+      class: "max-w-80 bg-white shadow-xs border border-alpha-black-50 border-solid py-4 px-4 rounded-[10px] text-sm font-medium flex gap-4 #{animation} group",
       data: {controller: "element-removal", action: actions },
       role: type == "error" ? "alert" : "status" do -%>
       <% base_class = "w-5 h-5 p-1 text-white flex shrink-0 items-center justify-center rounded-full" %>
       <%= type.in?(["error", "alert"]) ? lucide_icon("x", class: "#{base_class} bg-error") : lucide_icon("check", class: "#{base_class} bg-success") %>
-      <div class="flex flex-col gap-4">
-        <div class="flex flex-col gap-1">
+      <div class="flex flex-col">
+        <div class="flex flex-col">
```
Comment (evangelos-com) [inline app/views/shared/_notification.html.erb:14]: The gaps here were giving extra vertical white space, so I removed them.

File: app/views/shared/_notification.html.erb [hunk 4]
```diff
@@ -5,13 +5,13 @@
     <% actions = options[:auto_dismiss] ? "animationend->element-removal#remove" : "" %>
     <% animation = options[:auto_dismiss] ? "animate-[appear-then-fades_5s_300ms_both]" : "animate-[appear_5s_300ms_both]" %>
     <%= content_tag :div,
-      class: "max-w-80 bg-white shadow-xs border border-alpha-black-50 border-solid py-3 px-4 rounded-[10px] text-sm font-medium flex gap-4 #{animation}",
+      class: "max-w-80 bg-white shadow-xs border border-alpha-black-50 border-solid py-4 px-4 rounded-[10px] text-sm font-medium flex gap-4 #{animation} group",
```
Comment (evangelos-com) [inline app/views/shared/_notification.html.erb:8]: The close button will show when you hover over the outer most div of the notification.

File: app/views/shared/_notification.html.erb (hunk 2 above)
Comment (zachgoll) [inline app/views/shared/_notification.html.erb:38]:
```suggestion
            <%= lucide_icon "x", class: "w-5 h-5 p-0.5 hidden group-hover:inline-block border border-alpha-black-50 border-solid rounded-lg bg-white text-gray-400 cursor-pointer", data: { action: "click->element-removal#remove" } %>
```

**Review by zachgoll — APPROVED** [review]
Looks good, nice work!

**Pull request merged**